src/hera/volume_claim_gc.py

```diff
@@ -0,0 +1,36 @@
+from enum import Enum
+
+from argo_workflows.models import IoArgoprojWorkflowV1alpha1VolumeClaimGC
+from pydantic import BaseModel
+
+
+class VolumeClaimGCStrategy(Enum):
+    """A representation of the strategy to use when deleting volumes from completed workflows
+
+    Notes
+    -----
+        See sdk doc (IoArgoprojWorkflowV1alpha1VolumeClaimGC)
+    """
+
+    OnWorkflowCompletion = "OnWorkflowCompletion"
+    """delete volume claim when workflow is completed
+    """
+
+    OnWorkflowSuccess = "OnWorkflowSuccess"
+    """delete volume claim when workflow is successful
+    """
+
+
+class VolumeClaimGC(BaseModel):
```

Comment:
The [docs](https://argoproj.github.io/argo-workflows/fields/#volumeclaimgc) for volume claim GC suggest there's a single field at the moment, which is `strategy`. What do you think about using the `VolumeClaimGCStrategy` enum directly in a workflow `__init__` rather than wrapping it with a `VolumeClaimGC` `BaseModel`? I think `ImagePullPolicy` is similar to this - we use the enum directly. My decision boundary between using a base model and not using one is generally asking the question: "does this have more than 1 field and are there opportunities to construct logic on the fields?" If the answer is no, which I think is the case for image pull policy, GC, etc, then I suggest we don't use a `BaseModel`. Thoughts, @szdr?

Reply:
Thanks for your comment, I agree with the idea of using VolumeClaimGCStrategy directly in __init__ of workflow.